app/controllers/users_controller.rb

```diff
@@ -8,15 +8,26 @@ def new
   end
   
   def create
+    rollback = false
     @user = User.new(user_params)
     
-    if @user.save
-      check_for_invitation
-      AppMailer.delay.notify_on_user_signup(@user)
-      redirect_to sign_in_path
-    else
-      render :new
+    ActiveRecord::Base.transaction do
```

Comment:
Yea :) when I was doing this exercise myself I came up with Transaction too first :) very good thinking!